reactive/kubernetes_master.py

```diff
@@ -3658,11 +3696,73 @@ def configure_kubelet():
     kubernetes_common.configure_kubelet(
         dns_domain, dns_ip, registry, taints=taints, has_xcp=has_xcp
     )
+    set_state("kubernetes-master.config.created")
     service_restart("snap.kubelet.daemon")
-
+    set_state("kubernetes-master.label-config-required")
     set_flag("kubernetes-master.kubelet.configured")
 
 
+@when("config.changed.labels")
+def handle_labels_changed():
+    set_state("kubernetes-master.label-config-required")
+
+
+@when("kubernetes-master.label-config-required", "kubernetes-master.config.created")
+def apply_node_labels():
```

Comment:
I would wait for `kubernetes-master.apiserver.configured` here as well. We don't want to try to talk to an apiserver that's not up yet.

Reply:
Also `kubernetes-master.kubelet.configured`. We don't want to try to label a node that doesn't exist yet.

Reply:
If you make the `kubelet_kubeconfig_path` -> `kubeclientconfig_path` change I suggested in prior comments, I suspect you could also replace `kubernetes-master.config.created` here with `authentication.setup` since that's what signals the creation of the client kubeconfig [here](https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/blob/1853eb192065c27453b234c0b50ebf6093f4d790/reactive/kubernetes_master.py#L2231-L2237).

Reply:
Thanks for this review.  @kwmonroe was suggesting i push these changes down into kubernetes-master-worker-base such that they share much of this work.  Maybe I could still use similar reactive states to trigger the node label application